File: src/com/soywiz/flash/backend/swing/SwingTextureBase.scala
package com.soywiz.flash.backend.swing

import java.awt.image.BufferedImage

import com.soywiz.flash.backend.TextureBase

class SwingTextureBase(val image:BufferedImage) extends TextureBase {
  override val width: Int = image.getWidth
  override val height: Int = image.getHeight
}

File: backends/swing/src/com/soywiz/backend/swing/SwingTextureBase.java
package com.soywiz.backend.swing;

import com.soywiz.flash.backend.TextureBase;

import java.awt.image.BufferedImage;

public class SwingTextureBase implements TextureBase {
    final public BufferedImage image;

    public SwingTextureBase(BufferedImage image) {
        this.image = image;
    }

    @Override
    public int width() {
        return image.getWidth();
    }

    @Override
    public int height() {
        return image.getHeight();
    }
}
